sqlmesh/core/snapshot/definition.py

```diff
@@ -127,6 +127,23 @@ class SnapshotIntervals(PydanticModel, frozen=True):
     def snapshot_id(self) -> SnapshotId:
         return SnapshotId(name=self.name, identifier=self.identifier)
 
+    @classmethod
+    def from_snapshot(cls, snapshot: Snapshot, **kwargs: t.Any) -> SnapshotIntervals:
+        return cls(
+            **{
+                **{
+                    "name": snapshot.name,
+                    "identifier": snapshot.identifier,
+                    "version": snapshot.version,
+                    "intervals": snapshot.intervals.copy() if snapshot.intervals_ else [],
```

Comment:
If `snapshot.intervals_ ` is None that means we are copying a snapshot that's not hydrated and make it hydrated instead. Is this logic correct?

Reply:
Intervals are now empty list by default which removes this if/else logic: https://github.com/TobikoData/sqlmesh/pull/1018/commits/3323361b95251e980a2b3fea7244a3c23840e779